tests/restrictive_role_cf.yaml

```diff
@@ -0,0 +1,80 @@
+
```

Comment:
before you spend more effort on this, could you give a general idea on what you plan to implement? 
ideally all the necessary infra to run integration test is not implemented in dbt-athena, but in https://github.com/dbt-athena/dbt-athena-infra
then in dbt-athena CI we will assume role deployed somewhere else, and we could consider to also cover this use case ;)

Reply:
I think that supporting `native_drop` for iceberg is the way to got.
At some point, Athena will improve/enhance the limitations on removing all the s3 objects on bigger datasets, and if that happens having this feature backed in will be beneficial for use cases like yours, but at the same time could mean a change of a flag for users that want to benefit from it.

Reply:
Adopted native_drop suggestion

Reply:
Moved role defs to infra repo